lib/ssl/src/ssl_config.erl

```diff
@@ -87,6 +87,9 @@ group_pairs([#{private_key := #'DSAPrivateKey'{}} = Pair | Rest], #{dsa := DSA}
 group_pairs([#{private_key := #{algorithm := dss, engine := _}} = Pair | Rest], Group) ->
     Pairs = maps:get(dsa, Group),
     group_pairs(Rest, Group#{dsa => [Pair | Pairs]});
+group_pairs([#{private_key := #{algorithm := Alg, sign_fun := _}} = Pair | Rest], Group) ->
```

Comment:
Here my test would fail with `rsa_pss_rsae`, there is already a conversion in place for `dss` to `dsa` for the engine, I could do something similar and convert here, maybe to `rsa_pss_pss` or just `rsa`.

Reply:
This situation is a bit different.  DSA and DSS is just two names for the conceptually same thing. There is a theoretical difference being that one being  a reference  to the whole standard and the other to the algorithm. (I think this is mostly unfortunate and confusing, but it is at it is!) When it comes to rsa_pss_pss and rsa_pss_rsae it now depends on if the rsa key has any parameters or not.  Certificates using a certificate has a key that has no parameters must use the rsa identifier. So we can handle it as rsa or rsa_pss_pss.

Reply:
Ok, so I can just add a conversion to `rsa` for example. Since the key will be utilized by the user's function code anyway,  it shouldn't matter much in this case.